packages/react-html/src/components/HtmlUpdater.tsx

```diff
@@ -43,7 +43,7 @@ function updateOnClient(state: State) {
 
   const fragment = document.createDocumentFragment();
 
-  updateElement('meta', metas, fragment);
+  updateElement('meta', removeDuplicate(metas), fragment);
```

Comment:
Why are duplicate metas being added in the first place? Could we not use a Set or Map to get around this?

Reply:
We need to keep track of every metas added.

Imagine I have a `Layout` component that wraps my app and that will never unmount. It contains default meta for the description. Sometimes I will render a child component (e.g. AboutUsPage) that will want to override the description. If I go back to the homepage then the description added by `Layout`  should be back.

See https://github.com/Shopify/polaris-styleguide/pull/3776 for a real example